contracts/pair/src/contract.rs

```diff
@@ -772,19 +787,26 @@ fn get_current_weight(
     start_weight: Uint128,
     end_weight: Uint128,
     start_time: u64,
-    end_time: u64,
+    end_time: Option<u64>,
     block_time: u64,
 ) -> StdResult<Decimal256> {
     if block_time < start_time {
         return Err(StdError::generic_err("Sale has not started yet"));
     }
 
-    if block_time > end_time {
+    if block_time > end_time.unwrap() {
         return Err(StdError::generic_err("Sale has already finished"));
     }
 
     let start_weight_fixed = uint2dec(start_weight);
-    let time_diff = uint2dec(Uint128::from(end_time - start_time));
+
+    // AMM pair will not have an end time & must have equal start/end weights
+    if end_time == None {
+        let ratio = uint2dec(Uint128::zero());
+        return Ok(start_weight_fixed.sub(ratio));
+    }
```

Comment:
This code may be an error post-LBP. I suggest moving this logic to the top of the function.

Reply:
Will take a look at that.

Reply:
Moved that up, but can't see there being an issue with "Post-LBP" since all LBP pairs will have an `end_time`. Only AMM pairs would not have an `end_time`.